### DISCUSSION TITLE
Vercel + Google Cloud Storage get signed url return error 500

### DISCUSSION THREAD

**Discussion by ImBIOS** [vercel/community#1166] [Help] [answered]
### Summary

Vercel serverless function didn't support @google-cloud/storage package API or something.
Please refer to the Original Issue or Reference Issue if you want to debug or resolve this issue.

### Additional information

Original Issue: [ImBIOS/cardus-app #2](https://github.com/ImBIOS/cardus-app/issues/2)
Reference Issue: @leerob  [leerob/nextjs-gcp-storage #10](https://github.com/leerob/nextjs-gcp-storage/issues/10)

### Example

Sorry, I can't provide you my private bucket because it could be abused by spamming with big chunk or something.

**Comment by amyegan** [comment, marked as answer]
🤔 I'm not sure exactly what's going wrong here without digging into all of the details of your project, but here are some suggestions I found from others running into this kind of error.

- [PEM start key missing](https://stackoverflow.com/questions/63030755/error-error0909006cpem-routinesget-nameno-start-line-node)
- [mistake in `.env`](https://stackoverflow.com/questions/64826237/google-cloud-storage-call-fails-with-error0909006cpem-routinesget-nameno-st)
- [special characters in key](https://github.com/box/box-node-sdk/issues/502#issuecomment-615325419)
- [copied `\n` in key](https://stackoverflow.com/a/72308878)

Hopefully one of these solves it for you. If not, please let me know!

**Reply by ImBIOS** [reply]
I'd love the resources you're collected to solve my issue. I choose the easiest one for me:

Before:

```ts
process.env.PRIVATE_KEY
```

After:

```ts
process.env.PRIVATE_KEY?.replace(/\\n/gm, "\n")
```
